src/core/Points.tsx

```diff
@@ -121,4 +137,77 @@ const Point = React.forwardRef(({ children, ...props }, ref) => {
   )
 })
 
-export { Points, Point }
+/**
+ * Buffer implementation, relies on complete buffers of the correct number, leaves it to the user to update them
+ */
+type PointsBuffersProps = Omit<JSX.IntrinsicElements['points'], 'position'> & {
+  // a buffer containing all points position
+  positions: Float32Array
+  colors?: Float32Array
+  sizes?: Float32Array
+  // The size of the points in the buffer
+  stride?: 2 | 3
+}
+
+export const PointsBuffer = React.forwardRef<THREE.Points, PointsBuffersProps>(
+  ({ children, positions, colors, sizes, stride = 3, ...props }, forwardedRef) => {
+    const pointsRef = React.useRef<THREE.Points>(null!)
+
+    useFrame(() => {
+      const attr = pointsRef.current.geometry.attributes
+
+      attr.position.needsUpdate = true
+
+      if (colors) {
+        attr.color.needsUpdate = true
+      }
+
+      if (sizes) {
+        attr.size.needsUpdate = true
+      }
+    })
+
+    return (
+      <points matrixAutoUpdate={false} ref={mergeRefs([forwardedRef, pointsRef])} {...props}>
+        <bufferGeometry>
+          <bufferAttribute
+            attachObject={['attributes', 'position']}
+            count={positions.length / stride}
+            array={positions}
+            itemSize={stride}
+            usage={THREE.DynamicDrawUsage}
+          />
+          {colors && (
+            <bufferAttribute
+              attachObject={['attributes', 'color']}
+              count={colors.length / stride}
+              array={colors}
+              itemSize={stride}
+              usage={THREE.DynamicDrawUsage}
+            />
+          )}
+          {sizes && (
+            <bufferAttribute
+              attachObject={['attributes', 'size']}
+              count={sizes.length / stride}
+              array={sizes}
+              itemSize={stride}
+              usage={THREE.DynamicDrawUsage}
+            />
+          )}
+        </bufferGeometry>
+        {children}
+      </points>
+    )
+  }
+)
+
+export const Points = React.forwardRef<THREE.Points, PointsBuffersProps | PointsInstancesProps>(
+  (props, forwardedRef) => {
+    if ((props as PointsBuffersProps).positions instanceof Float32Array) {
+      return <PointsBuffer {...(props as PointsBuffersProps)} ref={forwardedRef} />
+    }
+
+    return <PointsInstances {...(props as PointsInstancesProps)} ref={forwardedRef} />
+  }
+)
```

Comment:
@drcmda the switch is based on positions being passed as a f32 array, feel free to change this to something else 🤷